### PR TITLE
New template for chained Data Factory pipeline

### PR DESCRIPTION
**Pull request by kskvarci**
### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use resourceGroup().location for resource locations
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/bp-checklist.md

- [ ] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

*
*
*

### CODE REVIEW

File: 101-data-factory-chained-copy-activities/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,370 @@
...
+      },
+      "azureSQLConnectionString": {
+        "type": "securestring",
+        "metadata": {
+          "description": "Connection string for Azure SQL Database"
+        }
+      }
+    },
+    "variables": {
+      "dataFactoryName": "[concat('CopyFromSQLToADFToAzureSQL', uniqueString(resourceGroup().id))]",
```
Comment (bmoore-msft) [inline 101-data-factory-chained-copy-activities/azuredeploy.json]: does this need to be globally unique or just unique within the RG?  i.e. can we remove uniqueString?

File: 101-data-factory-chained-copy-activities/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,370 @@
...
+            "dependsOn": [
+              "[variables('dataFactoryName')]",
+              "[variables('gatewayName')]"
+            ],
+            "properties": {
+              "type": "OnPremisesSqlServer",
+              "typeProperties": {
+                  "connectionString": "Placeholder-COnnectionString-Replace-Post-Deployment",
+                  "gatewayName": "[variables('gatewayName')]",
+                  "userName": "",
```
Comment (bmoore-msft) [inline 101-data-factory-chained-copy-activities/azuredeploy.json]: are these properties required?  they look to have bogus values and there is a connstr...

File: 101-data-factory-chained-copy-activities/azuredeploy.json (hunk 1 above)
Comment (kskvarci) [reply 101-data-factory-chained-copy-activities/azuredeploy.json]: Data Factory names are unique across Azure so the uniqueString is required: https://docs.microsoft.com/en-us/azure/data-factory/data-factory-naming-rules

File: 101-data-factory-chained-copy-activities/azuredeploy.json (hunk 2 above)
Comment (kskvarci) [reply 101-data-factory-chained-copy-activities/azuredeploy.json]: The property is required in order for the resource creation to succeed on deploy. There are explicit instructions in the README.md however to replace this value with an encrypted connection string post deployment (lines 38,39). The string cannot be encrypted until the deployment succeeds.

**Pull request merged**